LanguageFeatures/Subtyping/dynamic/generated/records_arguments_binding_A03_t03.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a type T0 is a subtype of a type T1 (written
+/// T0 <: T1) when:
+/// A record type A is a subtype of record type B iff they have same shape and
+/// the types of all fields of A are subtypes of the corresponding field types
+/// of B
+/// 
+/// @description Check that if type T0 is a record with the same shape that T1
+/// and the types of all fields of T0 is subtype of of the corresponding field
+/// types of T1 then T0 is subtype of T1. Check named fields only
+/// @author sgrekhov22@gmail.com
+///
+/// @description Check that if type T0 is a subtype of a type T1, then instance
+/// of T0 can be used as an argument of type T1. Test mixin members
+/// @author sgrekhov@unipro.ru
+///
+/// This test is generated from records_A03.dart and 
+/// arguments_binding_x03.dart.
+/// Don't modify it. If you want to change this test, change one of the files 
+/// above and then run generator.dart to regenerate the tests.
+
+
+import '../../utils/common.dart';
+
+// SharedOptions=--enable-experiment=records
+
+({int x, double y, String z}) t0Instance = (x: 42, y: 3.14, z: "");
+({String? z, num y, num x}) t1Instance = (y: 1, z: null, x: 0.1);
+
+const t1Default = (y: 1, z: null, x: 0.1);
+
+
+
+
+class ArgumentsBindingMixin1_t03 {
+  ({String? z, num y, num x}) m = t1Default;
+
+  void superTest(({String? z, num y, num x}) val) {}
+  void superTestPositioned(({String? z, num y, num x}) val, [({String? z, num y, num x}) val2 = t1Default]) {}
+  void superTestNamed(({String? z, num y, num x}) val, {({String? z, num y, num x}) val2 = t1Default}) {}
+  ({String? z, num y, num x}) get superGetter => m;
+  void set superSetter(({String? z, num y, num x}) val) {}
+}
+
+class ArgumentsBinding1_t03 extends Object with ArgumentsBindingMixin1_t03 {
+
+  test(dynamic t1, dynamic t2) {
+    superTest(t1);
+    superTestPositioned(t1);
+    superTestPositioned(t2, t1);
+    superTestNamed(t1);
+    superTestNamed(t2, val2: t1);
+    superSetter = t1;
+    m = t1;
+    superGetter;
+  }
+}
+
+class ArgumentsBindingMixin2_t03<X> {
```

Comment:
When we only have one of the cases, I think it's more important to test a `mixin` declaration than a `class` declaration used to derive a mixin. After all, the latter is probably going to be deprecated at some point (if we can do that without too much breakage).

Reply:
Agreed. It'll be a separate change

Reply:
OK